app/src/main/java/org/schabi/newpipe/fragments/BaseStateFragment.java

```diff
@@ -229,6 +231,12 @@ protected boolean onError(final Throwable exception) {
         } else if (ExceptionUtils.isNetworkRelated(exception)) {
             showError(getString(R.string.network_error), true);
             return true;
+        } else if (exception instanceof GeographicRestrictionException) {
+            showError(getString(R.string.georestricted_content), false);
+            return true;
+        } else if (exception instanceof SoundCloudGoPlusException) {
+            showError(getString(R.string.soundcloud_go_plus_content), false);
+            return true;
```

Comment:
You need to move this up. 
`GeographicRestrictionException` and `SoundCloudGoPlusException` both extend `ContentNotAvailableException`, thus both exceptions go into the first branch.